apis/network/v1beta1/netconfig_webhook.go

```diff
@@ -144,6 +172,79 @@ func valiateNetworks(
 	return allErrs
 }
 
+// valiateNetworksChanged
+// - validate if a network was removed
+// - subnet is still there
+// - cidr changed
+// - Vlan changed
+// - gateway changed
+func valiateNetworksChanged(
+	networks []Network,
+	oldNetworks []Network,
+	path *field.Path,
+) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	for oldNetIdx, _net := range oldNetworks {
+		path := path.Child("networks").Index(oldNetIdx)
+
+		// validate if networks are still there
+		f := func(n Network) bool {
+			return equality.Semantic.DeepEqual(n.Name, _net.Name)
+		}
+		netIdx := slices.IndexFunc(networks, f)
+		if netIdx < 0 {
+			// the network was removed
+			allErrs = append(allErrs, field.Invalid(path.Child("name"), _net.Name, fmt.Sprintf(errNetworkChanged, _net.Name)))
+			return allErrs
+		}
+
+		path = path.Child("subnets")
+		for oldSubnetIdx, _subnet := range _net.Subnets {
+			path := path.Index(oldSubnetIdx)
+
+			// validate if subnet is still there
+			f := func(n Subnet) bool {
+				return equality.Semantic.DeepEqual(n.Name, _subnet.Name)
+			}
+			subnetIdx := slices.IndexFunc(networks[netIdx].Subnets, f)
+			if subnetIdx < 0 {
+				allErrs = append(allErrs, field.Invalid(path.Child("name"), _net.Name, fmt.Sprintf(errSubnetParameterChanged, "name", _subnet.Name)))
+				return allErrs
+			}
+
+			// validate if cidr changed
+			if !equality.Semantic.DeepEqual(_subnet.Cidr, networks[netIdx].Subnets[subnetIdx].Cidr) {
+				allErrs = append(allErrs, field.Invalid(path.Child("cidr"), _net.Name, fmt.Sprintf(errSubnetParameterChanged, "cidr", _subnet.Cidr)))
+			}
+
+			// validate if Vlan changed
+			if !equality.Semantic.DeepEqual(_subnet.Vlan, networks[netIdx].Subnets[subnetIdx].Vlan) {
+				vlan := ""
+				if _subnet.Vlan != nil {
+					vlan = strconv.Itoa(*_subnet.Vlan)
+				}
+				allErrs = append(allErrs, field.Invalid(path.Child("vlan"), _net.Name, fmt.Sprintf(errSubnetParameterChanged, "vlan", vlan)))
+			}
+
+			// validate if gateway changed
+			if !equality.Semantic.DeepEqual(_subnet.Gateway, networks[netIdx].Subnets[subnetIdx].Gateway) {
+				gateway := ""
+				if _subnet.Gateway != nil {
+					gateway = *_subnet.Gateway
+				}
+				allErrs = append(allErrs, field.Invalid(path.Child("vlan"), _net.Name, fmt.Sprintf(errSubnetParameterChanged, "vlan", gateway)))
+			}
+
+		}
+		if len(allErrs) > 0 {
+			return allErrs
+		}
```

Comment:
Do we want to `return` here if there are errors for this network?  Just curious if it would make sense to check the other networks and report their potential errors as well.

Reply:
I'll change it to have the same as on create. I thought it might make sense to report issues per network as it might be easier to understand. But now see that its different from create

Reply:
done